pyscraper/sp_2024/convert.py

```diff
@@ -0,0 +1,180 @@
+    dest_path.parent.mkdir(parents=True, exist_ok=True)
+
+    id_factory = IDFactory(committee_slug=committee_slug, iso_date=iso_date)
+
+    # create a new major heading from the general description
+    major_heading = etree.Element("major_heading")
+
+    major_heading.set("id", id_factory.get_next_major_id())
+    major_heading.set("url", url)
+    major_heading.set("nospeaker", "True")
+    major_heading.text = f"{title} {date_str}"
+    root.append(major_heading)
+
+    # iterate through the agenda_items
+    for item in source.iter("agenda_item"):
+        # create a new major_heading from the agenda_item information
+        major_heading = etree.Element("major_heading")
+        major_heading.set("id", id_factory.get_next_major_id())
+        major_heading.set("url", item.get("url"))
+        major_heading.set("nospeaker", "True")
+        major_heading.text = item.get("title")
+        root.append(major_heading)
+
+        # iterate through the agenda_item's subitems
+        # if item is a speech or a division, get the new informaton and bring across
+        previous_speech = None
+        for subitem in item.find("parsed"):
+            if subitem.tag == "speech":
+                speaker_name = subitem.get("speaker_name")
+                person_id = get_unique_person_id(speaker_name, iso_date)
+                if person_id is None and verbose:
+                    print(f"Could not find person id for {speaker_name}")
+                speech = etree.Element("speech")
+                speech.set("id", id_factory.get_next_minor_id())
+                speech.set("url", subitem.get("speech_url"))
+                speech.set("speakername", speaker_name)
+                speech.set("person_id", person_id or "Unknown")
+                for child in subitem:
+                    speech.append(child)
+                root.append(speech)
+                previous_speech = speech
+
+            elif subitem.tag == "division":
+                # get previous sibling of the subitem to get the speech info
+
+                if previous_speech is None:
+                    raise ValueError("Division without a previous speech")
+
+                division = etree.Element("division")
+                division.set("id", id_factory.get_next_minor_id())
+                division.set("url", previous_speech.get("url"))
+                division.set("divdate", iso_date)
+                division.set("divnumber", subitem.get("divnum"))
+                division.set("nospeaker", "True")
+                for child in subitem:
+                    division.append(child)
+                root.append(division)
+
+    # for all mspnames elements, we need to create an ID property
+    for mspname in root.iter("mspname"):
+        person_name = mspname.text
+        person_id = is_member_vote(person_name, iso_date)
+        if person_id is None:
+            print(f"Could not find person id for {person_name}")
```

Comment:
Ah, as in it specifies who the presiding officer is once a day - but because we're parsing in bits, this is being lost. 

ID matching would be ideal - but we still need name matching for parsing votes. 

Alternative (easier??) approach - "the convener" items do have the ID associated with them. We could pass in the name and ID, it internally stores the name against the ID if it finds a match - otherwise processes the name.

Reply:
Not sure what you mean by "the convener items" sorry. But yes, if we pass in ID and name, and it remembers based upon the ID, that would catch anything like this.

Reply:
This should fix that, seems a lot happier with conveners and presiding officers. https://github.com/mysociety/parlparse/pull/172/commits/20cc60c643994c98ce4c367ab28d5e9ee6b1de94